File: TypeScript/Problem214.ts
function shortestPalindrome(s: string): string {
    let reversed = '';
    
    for(let i = s.length-1; i >= 0; i--){
        reversed += s[i];
    }

    for(let i = 0; i < s.length; i++){
        let prefix = s.substring(0, s.length - i);
        let suffix = reversed.substring(i);

        if (prefix == suffix){
            return reversed.substring(0, i) + s;
        }
    }

    return s;
};

File: Java/Problem214.java
class Solution {
    public String shortestPalindrome(String s) {
        int StringSize = s.length();
        String ReversedString = "";

        // Reverse the String
        for (int i = 0; i<StringSize ; i++){
            ReversedString = s.charAt(i) + ReversedString;
        }
        //System.out.println(ReversedString);
        // Works

        for (int i = 0 ; i<StringSize ; i++){
            String Prefix = s.substring(0,StringSize - i);
            String Suffix = ReversedString.substring(i);

            if (Prefix.equals(Suffix)){
                System.out.println("here");
                return ReversedString.substring(0,i) + s;
            }
        }
        return s;
    }
}